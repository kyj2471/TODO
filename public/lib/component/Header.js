'use strict';Object.defineProperty(exports,'__esModule',{value:!0});var _link=require('next/link'),_link2=_interopRequireDefault(_link),_react=require('react'),_react2=_interopRequireDefault(_react),_reactRedux=require('react-redux'),_rootAction=require('../store/modules/rootAction'),actions=_interopRequireWildcard(_rootAction),_moment=require('moment'),_moment2=_interopRequireDefault(_moment),_Modal=require('./Modal'),_Modal2=_interopRequireDefault(_Modal),_UserModal=require('./UserModal'),_UserModal2=_interopRequireDefault(_UserModal),_styledComponents=require('styled-components'),_styledComponents2=_interopRequireDefault(_styledComponents);function _interopRequireWildcard(a){if(a&&a.__esModule)return a;var b={};if(null!=a)for(var c in a)Object.prototype.hasOwnProperty.call(a,c)&&(b[c]=a[c]);return b.default=a,b}function _interopRequireDefault(a){return a&&a.__esModule?a:{default:a}}var NavHead=_styledComponents2.default.div.withConfig({displayName:'Header__NavHead',componentId:'ktrsbb-0'})(['display:flex;justify-content:space-around;padding-top:30px;max-width:450px;margin:0 auto;img{width:50px;height:50px;}']),Header=function(a){var b=a.myModal,c=a.setMyModal,d=a.profileModal,e=a.setProfileModal,f=a.weather,g=a.getWeather,h=a.user,i=a.getUser;return(0,_react.useEffect)(function(){g(),i()},[]),<NavHead>
      <div onClick={function a(){c(!b)}}>날씨</div>
      {b&&<_Modal2.default myModal={b}setMyModal={c}weather={f}getWeather={g}/>}

      <div onClick={function a(){e(!d)}}>회원</div>
      {d&&<_UserModal2.default profileModal={d}setProfileModal={e}user={h}getUser={i}/>}

      <_link2.default href="/Image">
        <div>{(0,_moment2.default)().format('YYYY-MM-DD')}</div>
      </_link2.default>
    </NavHead>},mapStateToProps=function(a){var b=a.weather,c=a.getWeather,d=a.user,e=a.getUser;return{weather:b,getWeather:c,user:d,getUser:e}},mapDispatchToProps=function(a){return{getWeather:function c(b){return a(actions.getWeather(b))},getUser:function c(b){return a(actions.getUser(b))}}};exports.default=(0,_reactRedux.connect)(mapStateToProps,mapDispatchToProps)(Header);