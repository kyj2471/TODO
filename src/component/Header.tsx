import Link from 'next/link';
import React, { useEffect } from 'react';
import { connect } from 'react-redux';
import * as actions from '../store/modules/rootAction';
import moment from 'moment';
import Modal from './Modal';
import UserModal from './UserModal';
import styled from 'styled-components';
import { WeatherType } from '../store/interface/interface';
import { UserType } from '../store/interface/interface';

const NavHead = styled.div`
  display: flex;
  justify-content: space-around;
  padding-top: 30px;
  max-width: 450px;
  margin: 0 auto;

  img {
    width: 50px;
    height: 50px;
  }
`;

interface HeaderProps {
  myModal: boolean;
  setMyModal: React.Dispatch<React.SetStateAction<boolean>>;
  profileModal: boolean;
  setProfileModal: React.Dispatch<React.SetStateAction<boolean>>;
  weather: WeatherType;
  getWeather: any;
  user: UserType;
  getUser: any;
}

const Header = ({
  myModal,
  setMyModal,
  profileModal,
  setProfileModal,
  weather,
  getWeather,
  user,
  getUser
}: HeaderProps) => {
  const handleModal = () => {
    setMyModal(!myModal);
  };

  const handleProfileModal = () => {
    setProfileModal(!profileModal);
  };

  useEffect(() => {
    getWeather();
    getUser();
  }, []);

  return (
    <NavHead>
      <div onClick={handleModal}>날씨</div>
      {myModal && <Modal setMyModal={setMyModal} weather={weather} />}

      <div onClick={handleProfileModal}>회원</div>
      {profileModal && (
        <UserModal setProfileModal={setProfileModal} user={user} />
      )}

      <Link href="/Image">
        <div>{moment().format('YYYY-MM-DD')}</div>
      </Link>
    </NavHead>
  );
};
const mapStateToProps = ({
  weather,
  getWeather,
  user,
  getUser
}: HeaderProps) => ({
  weather,
  getWeather,
  user,
  getUser
});

const mapDispatchToProps = (dispatch: any) => ({
  getWeather: (weather: HeaderProps) => dispatch(actions.getWeather(weather)),
  getUser: (user: HeaderProps) => dispatch(actions.getUser(user))
});

export default connect(mapStateToProps, mapDispatchToProps)(Header);
