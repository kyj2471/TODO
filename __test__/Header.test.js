import React from 'react';
import Header from '../src/component/Header.tsx';
import configureStore from 'redux-mock-store';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import { render } from '@testing-library/react';

const mockStore = configureStore([]);

describe('please...', () => {
  let store;
  let component;
  beforeEach(() => {
    store = mockStore({
      state: 'plz'
    });
    component = renderer.create(
      <Provider store={store}>
        <Header />
      </Provider>
    );
  });
  it('you? can?', () => {
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
