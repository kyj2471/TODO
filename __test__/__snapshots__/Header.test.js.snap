// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`please... you? can? 1`] = `
<div
  className="sc-jRQAMF chXDxc"
>
  <div
    onClick={[Function]}
  >
    날씨
  </div>
  <div
    onClick={[Function]}
  >
    회원
  </div>
  <div
    onClick={[Function]}
    onMouseEnter={[Function]}
  >
    2021-05-27
  </div>
</div>
`;
